proto/attestation/reference_value.proto

```diff
@@ -64,39 +65,41 @@ message VerifyLogEntry {
 
 // Verifies that the field contains at least one of the given digests.
 // No checks are performed if this is empty.
-message Measurements {
+message Digests {
   repeated RawDigest digests = 1;
 }
 
-message VerifyLinuxKernel {
-  // Each field is in a 1:1 correspondence to a CBOR key.
-  optional VerifyLogEntry kernel_binary = 1;
-  optional VerifyLogEntry kernel_cmd_line = 2;
-  optional VerifyLogEntry acpi = 3;
-}
+// Reference values of the kernel layer, as measured by stage0.
+message KernelLayerReferenceValues {
+  // Verifies the kernel binary based on endorsement.
+  optional LogEntryVerification log_entry = 1;
 
-message VerifyOakRestrictedKernel {
-  // TBD, but similar to Linux kernel
-}
+  // Explicitly verifies digests in addition to log entry verification.
```

Comment:
I still don't get the semantics of this; if we know the list of digests ahead of time, what's the point of checking log entries too? It seems these checks are mutually exclusive.

Reply:
Log entry verification is based on endorsement from the server, explicit digests come from the client which are different sources. I don't think the overhead of a oneof is warranted, as it would only exclude the possibility to demand both checks. We could move the `Digests` field into LogEntryVerification (and probably rename LogEntryVerification), so it would be allowed systematically (still no oneof). But it really depends on how this will be used in practice.

Reply:
My question is: why would anyone hardcode an expected digest on the client and also require that the client at runtime checks it was in a log? If they hardcoded it they would have checked whatever they wanted about it out of band already.

Reply:
Put differently, verifying the digests when there is a log entry indicates mistrust in Transparent Release. Someone expects that it maybe compromised. Since it is our own effort which we want to be trusted, maybe shouldn't provide some kind of workaround? This would indeed point to removing the explicit digest verification (everywhere, there are three instances right now).